tools/topology2/sof-tgl-sdw-max98373-rt5682-4ch.conf

```diff
@@ -0,0 +1,398 @@
+<include/pipelines/pipeline-kpb-vol-capture.conf>
+<include/pipelines/pipeline-smart-amp-playback.conf>
+<include/pipelines/pipeline-smart-amp-ref-capture.conf>
+<include/pipelines/pipeline-detect.conf>
+<include/common/connection.conf>
+<include/common/endpoint.conf>
+<include/components/dai.conf>
+<include/components/detect.conf>
+<include/components/detect_data.conf>
+<include/components/smart_amp.conf>
+<include/components/smart_amp_data.conf>
+<include/dais/hda.conf>
+<include/dais/alh.conf>
+<include/dais/dmic.conf>
+
+#
+# Pipeline definitions
+#
+# Pipeline ID:1 PCM ID: 0
+Object.pipeline-volume-playback."1.0" {
+	pcm_name		"Jack Out"
+	format			"s32le"
+	channels		2
+	rate			48000
+	volume_ctl_name	"1 Master Playback Volume"
+}
+
+# Pipeline ID:2 PCM ID: 1
+Object.pipeline-volume-capture."2.1" {
+	pcm_name		"Jack in"
+	format			"s32le"
+	channels		2
+	rate			48000
+	volume_ctl_name	"2 Master Capture Volume"
+}
+
+# Pipeline ID:3 PCM ID: 2
+Object.pipeline-smart-amp-playback."3.2" {
+	pcm_name		"Speaker"
+	format			"s32le"
+	channels		2
+	rate			48000
+	smart_amp_class	"smart_amp"
+	smart_amp_config	"dummy_amp_config"
+	smart_amp_model	"dummy_amp_model"
+	smart_amp_model_get	"dummy_amp_model_get"
+}
+
+# Pipeline ID:4 PCM ID: 3
+Object.pipeline-smart-amp-ref-capture."4.3" {
+	pcm_name	"Amplifier Reference"
+	format		"s32le"
+	channels	2
+	rate		48000
+}
+
+# Pipeline ID:6 PCM ID: 5
+Object.pipeline-volume-playback."6.5" {
+	pcm_name		"HDMI1"
+	format			"s32le"
+	channels		2
+	rate			48000
+	volume_ctl_name	"6 Master Playback Volume"
+}
+
+# Pipeline ID:7 PCM ID: 6
+Object.pipeline-volume-playback."7.6" {
+	pcm_name		"HDMI2"
+	format			"s32le"
+	channels		2
+	rate			48000
+	volume_ctl_name	"7 Master Playback Volume"
+}
+
+# Pipeline ID:8 PCM ID: 7
+Object.pipeline-volume-playback."8.7" {
+	pcm_name		"HDMI3"
+	format			"s32le"
+	channels		2
+	rate			48000
+	volume_ctl_name	"8 Master Playback Volume"
+}
+
+# Pipeline ID:9 PCM ID: 8
+Object.pipeline-volume-playback."9.8" {
+	pcm_name		"HDMI4"
+	format			"s32le"
+	channels		2
+	rate			48000
+	volume_ctl_name	"9 Master Playback Volume"
+}
+
+# Pipeline ID:11 PCM ID: 10
+Object.pipeline-passthrough-capture."11.10" {
+	pcm_name	"DMIC"
+	format		"s32le"
+	channels	4
+	rate		48000
+}
+
+# Pipeline ID:12 PCM ID: 12
+Object.pipeline-kpb-vol-capture."12.12" {
+	pcm_name		"BufferedMic"
+	format			"s32le"
+	channels		2
+	rate			16000
+	period			20000
+	volume_ctl_name	"9 KWD Capture Volume"
+}
+
+# Pipeline ID:13
+Object.pipeline-detect."13" {
+	format			"s24le"
+	channels		2
+	rate			16000
+	period			20000
+	stream_name		"BufferedMic.capture.12"
+	detect_sink_name	"Detect Sink 13"
+	detect_object_class	"detect"
+	detect_config		"dummy_detect_config"
+	detect_model		"dummy_detect_model"
+}
+
+#
+# List of all DAIs
+#
+#ALH Index: 2, Direction: playback
+Object.ALH."2.2.playback" {
+	dai_name		"SDW0-Playback"
+	id 			0
+	default_hw_config	0
+	format			"s24le"
+	rate			48000
+	ch			2
+	hw_config."0" {}
```

Comment:
Is the SSP config somewhere ?

Reply:
This file is only for the SDW topology but the SSP configs are in the sof-tlg-max98373-rt5682.conf file.